packages/grafana-ui/src/components/AutoSaveInput/AutoSaveInput.tsx

```diff
@@ -0,0 +1,115 @@
+import { debounce } from 'lodash';
+import React, { useCallback, useMemo, useRef } from 'react';
+
+import { Field } from '../Forms/Field';
+import { InlineToast } from '../InlineToast/InlineToast';
+import { Input, Props as InputProps } from '../Input/Input';
+
+/**
+1.- Use the Input component as a base
+2.- Just save if there is any change and when it loses focus
+3.- Set the loading to true while the backend is saving
+4.- Be aware of the backend response. If there is an error show a proper message and return the focus to the input.
+5.- Add aria-live="polite" and check how it works in a screen-reader.
+Debounce instead of working with onBlur?
+import debouncePromise from 'debounce-promise';
+or
+import { debounce} from 'lodash';
+ */
+
+export interface Props extends InputProps {
+  //Function to be run onBlur or when finishing writing
+  onFinishChange: (inputValue: string | number | readonly string[] | undefined) => Promise<void>;
+  customErrorMessage?: string;
+}
+
+const SHOW_SUCCESS_DURATION = 2 * 1000;
+
+export const AutoSaveInput = React.forwardRef<HTMLInputElement, Props>((props) => {
+  const {
+    defaultValue = '',
+    className,
+    addonAfter,
+    addonBefore,
+    prefix,
+    suffix,
+    invalid,
+    loading,
+    onFinishChange,
+    customErrorMessage,
+    ...restProps
+  } = props;
+  const [isLoading, setIsLoading] = React.useState(false);
+  const [showSuccess, setShowSuccess] = React.useState(false);
+  const [showError, setShowError] = React.useState(false);
+  const [showErrorMessage, setShowErrorMessage] = React.useState(customErrorMessage || 'Error saving this value');
```

Comment:
Looks like combining this to one state object would simplify the code a bit (not so many separate set state actions)